numpy/lib/shape_base.py

```diff
@@ -134,9 +134,9 @@ def apply_along_axis(func1d, axis, arr, *args, **kwargs):
         buff = res.__array_prepare__(buff)
 
     # save the first result, then compute and save all remaining results
-    buff[ind0] = res
+    buff[ind0] = res[()]
```

Comment:
Sure is possible, though you must be sure you have an array. Otherwise, object dtype might blow up in your face.